src/app/ngrx/actions/castle.action.creator.ts

```diff
@@ -1,29 +1,91 @@
 import { Action } from '@ngrx/store';
 import { ActionTypes } from './action.types';
-import { EpisodeModel, PodcastModel } from '../';
+import { EpisodeModel, PodcastModel, MetricsType, IntervalModel } from '../';
 
-export interface CastlePodcastMetricsPayload {
-  podcast: PodcastModel;
+export interface CastlePodcastMetricsLoadPayload {
+  seriesId: number;
+  feederId: string;
+  metricsType: MetricsType;
+  interval: IntervalModel;
+  beginDate: Date;
+  endDate: Date;
+}
+
+export class CastlePodcastMetricsLoadAction {
+  readonly type = ActionTypes.CASTLE_PODCAST_METRICS_LOAD;
+
+  constructor(public payload: CastlePodcastMetricsLoadPayload) {}
+}
+
+export interface CastlePodcastMetricsSuccessPayload {
+  seriesId: number;
+  feederId: string;
   metricsPropertyName: string;
   metrics: any[][];
 }
 
-export class CastlePodcastMetricsAction implements Action {
-  readonly type = ActionTypes.CASTLE_PODCAST_METRICS;
+export class CastlePodcastMetricsSuccessAction implements Action {
+  readonly type = ActionTypes.CASTLE_PODCAST_METRICS_SUCCESS;
 
-  constructor(public payload: CastlePodcastMetricsPayload) {}
+  constructor(public payload: CastlePodcastMetricsSuccessPayload) {}
 }
 
-export interface CastleEpisodeMetricsPayload {
-  episode: EpisodeModel;
+export interface CastlePodcastMetricsFailurePayload {
+  seriesId: number;
+  feederId: string;
+  error: any;
+}
+
+export class CastlePodcastMetricsFailureAction implements Action {
+  readonly type = ActionTypes.CASTLE_PODCAST_METRICS_FAILURE;
+
+  constructor(public payload: CastlePodcastMetricsFailurePayload) {}
+}
+
+export interface CastleEpisodeMetricsLoadPayload {
```

Comment:
I should also explain why I moved away from using the models on these payloads. That's the TypeScript issue that requires we wrap our switch cases in those redundant if instance of checks. I realized that's what Todd Motto has been doing to get around it. The generic Action type has a payload, so you can destructure the payload in the reducer to get at the properties. Any deeper than one level past the payload, and you've got to use all those dumb instance of checks.